public/app/features/explore/TraceView/components/utils/filter-spans.tsx

```diff
@@ -109,25 +109,16 @@ const getTagMatches = (spans: TraceSpan[], tags: Tag[]) => {
       // match against every tag filter
       return tags.every((tag: Tag) => {
         if (tag.key && tag.value) {
-          if (
-            span.tags.some((kv) => checkKeyAndValueForMatch(tag, kv)) ||
-            span.process.tags.some((kv) => checkKeyAndValueForMatch(tag, kv)) ||
-            (span.logs && span.logs.some((log) => log.fields.some((kv) => checkKeyAndValueForMatch(tag, kv)))) ||
-            (span.kind && tag.key === KIND && tag.value === span.kind) ||
-            (span.statusCode !== undefined &&
-              tag.key === STATUS &&
-              tag.value === SpanStatusCode[span.statusCode].toLowerCase()) ||
-            (span.statusMessage && tag.key === STATUS_MESSAGE && tag.value === span.statusMessage) ||
-            (span.instrumentationLibraryName &&
-              tag.key === LIBRARY_NAME &&
-              tag.value === span.instrumentationLibraryName) ||
-            (span.instrumentationLibraryVersion &&
-              tag.key === LIBRARY_VERSION &&
-              tag.value === span.instrumentationLibraryVersion) ||
-            (span.traceState && tag.key === TRACE_STATE && tag.value === span.traceState) ||
-            (tag.key === ID && tag.value === span.spanID)
-          ) {
+          if (tag.operator=== '=' && checkKeyValConditionForMatch(tag, span)) {
             return getReturnValue(tag.operator, true);
+          } else if (tag.operator === '=~' && checkKeyValConditionForRegex(tag, span)) {
+            return getReturnValue(tag.operator, false);
```

Comment:
```suggestion
          } else if (tag.operator === '=~' && checkKeyValConditionForRegex(tag, span)) {
            return getReturnValue(tag.operator, true);
```

Will also have to modify `getReturnValue` right?

Reply:
updated the getReturnValue method but not sure if it's needed now. Earlier, the method had binary operation supporting = or != but now, it supports 4 operations. Can you please check if this method is required? Thanks